File: java/placedOrdersStatistics/src/entities/Product.java
package entities;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Product {
    private String name;
    private String category;
    private double weight;
    private double price;
    private LocalDate creationDate;

    public Product(String name, String category, double weight, double price, LocalDate creationDate) {
        this.name = name;
        this.category = category;
        this.weight = weight;
        this.price = price;
        this.creationDate = creationDate;
    }

    public String getName() {
        return name;
    }

    public void setName(String nameProduct) {
        this.name = nameProduct;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public LocalDate getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDate creationDate) {
        this.creationDate = creationDate;
    }

    @Override
    public String toString() {
        return "Product{" +
                "name='" + name + '\'' +
                ", category='" + category + '\'' +
                ", weight=" + weight +
                ", price=" + price +
                ", creationDate=" + creationDate +
                '}';
    }
}

File: scala/src/main/scala/entity/Product.scala
package entity

import java.time.LocalDate

class Product(
               //private var id: Long,
             private var name: String,
             private var category: String,
             private var weight: Double,
             private var price: Double,
             private var creationDate: LocalDate
              ){

  //def getId: Long = id

  def getName: String = name

  def setName(newName: String): Unit = {
    name = newName
  }

  def getCategory: String = category

  def setCategory(newCategory: String): Unit = {
    category = newCategory
  }

  def getWeight: Double = weight

  def setWeight(newWeight: Double): Unit = {
    weight = newWeight
  }

  def getPrice: Double = price

  def setPrice(newPrice: Double): Unit = {
    price = newPrice
  }

  def getCreationDate: LocalDate = creationDate

  def setCreationDate(newCreationDate: LocalDate): Unit = {
    creationDate = newCreationDate
  }

  override def toString: String = {
    s"Nome: $name, Categoria: $category, Peso: $weight, Preço: $price, Data de criação: $creationDate"
  }


}
